### PR TITLE
Language agnostic additional data types

### PR DESCRIPTION
**Pull request by GregHib**
AB => This is now related to AppFlow v3 work, pending further definition.

### CODE REVIEW

File: flow-base-api/src/main/java/com/aevi/sdk/flow/model/AdditionalData.java [hunk 1]
```diff
@@ -433,4 +439,29 @@ public String toJson() {
     public static AdditionalData fromJson(String json) {
         return JsonConverter.deserialize(json, AdditionalData.class);
     }
+
+    public static Class<?> getClassFromType(String type) {
+        switch (type) {
+            case "Byte":
```
Comment (brettwold) [inline flow-base-api/src/main/java/com/aevi/sdk/flow/model/AdditionalData.java:445]: lets make these lower case please to be consistent

Comment (brettwold) [reply flow-base-api/src/main/java/com/aevi/sdk/flow/model/AdditionalData.java:445]: Also should be statics somewhere

**Review by brettwold — CHANGES_REQUESTED** [review]

**Review by brettwold — COMMENTED** [review]
Ah, this reminds me... we need to get rid of the double data object structure in here too if possible

**Comment by GregHib** [comment]
> Ah, this reminds me... we need to get rid of the double data object structure in here too if possible

Not without keeping backwards compatibility that I can see 🙁 

Ways I looked at doing it:

# 1

Serialize/deserialize the map directly and un/wrap manually in to/fromJson
Would require a new JsonConverter method which accepts a gson TypeToken to allow deserializing a map (exposes gson)
Wouldn't be easily be able to keep backwards compatibility; could deserialize as a JsonElement first to check if it has the data field and re-call and wrap when appropriate or trying and falling back after an exception. 🤮 

# 2

Hide the data field using `@Expose` and add a getter with `@JsonConverter.ExposeMethod(value = "")` and tweak ExposedMethod to add fields to the parent element.

However for `@Expose` to work gson requires `excludeFieldsWithoutExposeAnnotation` added to the builder, meaning all other fields would need to be marked with `@Expose` as well, which would also break backwards compatibility.

# 3

Switch Gson for Jackson and use the nice and easy `@JsonUnwrap` 😄 
But that'd break all the existing annotations 🤦

**Comment by GregHib** [comment]
The problem with using just JSON types is anyone currently storing & retriving an integer (or similar) with the auto-casting `Object getValue(String, Object...)`, switching to a version that seralizes that as a "number" type will fail to deserialize trying to cast it from Double to Int. It can be fixed switching to the generic type method but it'd be a breaking change none-the-less

**Pull request closed without merging**